blueprints/data-solutions/bq-ml/main.tf

```diff
@@ -0,0 +1,259 @@
+  source          = "../../../modules/project"
+  name            = var.project_id
+  parent          = try(var.project_create.parent, null)
+  billing_account = try(var.project_create.billing_account_id, null)
+  project_create  = var.project_create != null
+  prefix          = var.project_create == null ? null : var.prefix
+  services = [
+    "aiplatform.googleapis.com",
+    "bigquery.googleapis.com",
+    "bigquerystorage.googleapis.com",
+    "bigqueryreservation.googleapis.com",
+    "compute.googleapis.com",
+    "ml.googleapis.com",
+    "notebooks.googleapis.com",
+    "servicenetworking.googleapis.com",
+    "stackdriver.googleapis.com",
+    "storage.googleapis.com",
+    "storage-component.googleapis.com"
+  ]
+
+  shared_vpc_service_config = local.shared_vpc_project == null ? null : {
+    attach       = true
+    host_project = local.shared_vpc_project
+  }
+
+  service_encryption_key_ids = {
+    compute = [try(local.service_encryption_keys.compute, null)]
+    bq      = [try(local.service_encryption_keys.bq, null)]
+    storage = [try(local.service_encryption_keys.storage, null)]
+  }
+  service_config = {
+    disable_on_destroy = false, disable_dependent_services = false
+  }
+}
+
+###############################################################################
+#                                Networking                                   #
+###############################################################################
+
+module "vpc" {
+  source     = "../../../modules/net-vpc"
+  count      = local.use_shared_vpc ? 0 : 1
+  project_id = module.project.project_id
+  name       = "${var.prefix}-vpc"
+  subnets = [
+    {
+      ip_cidr_range = "10.0.0.0/20"
+      name          = "${var.prefix}-subnet"
+      region        = var.region
+    }
+  ]
+}
+
+module "vpc-firewall" {
+  source     = "../../../modules/net-vpc-firewall"
+  count      = local.use_shared_vpc ? 0 : 1
+  project_id = module.project.project_id
+  network    = module.vpc.0.name
+  default_rules_config = {
+    admin_ranges = ["10.0.0.0/20"]
+  }
+  ingress_rules = {
+    #TODO Remove and rely on 'ssh' tag once terraform-provider-google/issues/9273 is fixed
+    ("${var.prefix}-iap") = {
+      description   = "Enable SSH from IAP on Notebooks."
+      source_ranges = ["35.235.240.0/20"]
+      targets       = ["notebook-instance"]
+      rules         = [{ protocol = "tcp", ports = [22] }]
+    }
+  }
+}
+
+module "cloudnat" {
+  source         = "../../../modules/net-cloudnat"
+  count          = local.use_shared_vpc ? 0 : 1
+  project_id     = module.project.project_id
+  name           = "${var.prefix}-default"
+  region         = var.region
+  router_network = module.vpc.0.name
+}
+
+resource "google_project_iam_member" "shared_vpc" {
+  count   = local.use_shared_vpc ? 1 : 0
+  project = var.network_config.host_project
+  role    = "roles/compute.networkUser"
+  member  = "serviceAccount:${module.project.service_accounts.robots.notebooks}"
```

Comment:
can this be done via the project module instead?

Reply:
This is in the host_project, not in the one created in this blueprint.